src/algebra/big_operators/finprod.lean

```diff
@@ -700,4 +700,16 @@ lemma finsum_mul {R : Type*} [semiring R] (f : α → R) (r : R)
   (∑ᶠ a : α, f a) * r = ∑ᶠ a : α, f a * r :=
 (add_monoid_hom.mul_right r).map_finsum h
 
+@[to_additive]
+lemma finprod_emb_domain (f : α ↪ β) [decidable_pred (∈ set.range f)] (g : α → M) :
+  ∏ᶠ (b : β), (if h : b ∈ set.range f then g (classical.some h) else 1) = ∏ᶠ (a : α), g a :=
```

Comment:
The left hand side looks rather complicated and specialized. Do you think this could be replaced with the indicator function on the range of `f`? Also, I can imagine that someone will want to `rw` with this lemma, but only has an ordinary function that is `function.injective`. Would you mind also providing that version?

Reply:
I can definitely do an injective version, but I'm having trouble determining what other form this could take. It's not equivalent to just an indicator function, because the thing being summed over depends on `h : b ∈ set.range f`. Perhaps a general `classical.some` lemma is the greatest generality?

Reply:
I've made a helper lemma that does most of the work, by allowing you to convert between a `finprod_mem` that depends on the actual `mem` statement and one that does not, but I've left the original statement put. I expect it will find more use than just this, because if we end up refactoring `finsupp` to use `finsum`, this is the critical idea behind summing `finsupp.emb_domain`.